sectionproperties/pre/pre.py

```diff
@@ -505,7 +511,100 @@ def remove_duplicate_facets(self):
+    :param holes: List of points *(x, y)* defining the locations of holes within the cross-section.
+        If there are no holes, provide an empty list [].
+    :type holes: list[list[float, float]]
+    :param control_points: A list of points *(x, y)* that define different regions of the
+        cross-section. A control point is an arbitrary point within a region enclosed by facets.
+    :type control_points: list[list[float, float]]
+    :param tolerance: minimum permissable distance from facet
+    :type tolerance: float
+
+    General method was adapted from:
+    https://stackoverflow.com/a/54442561
+    """
+    # make sure points is a numpy array
+    points = np.asarray(points)
+    check_points = np.array(holes + control_points)
+
+    # select the starting and ending points of each facet
+    facet_points = points[facets]
+    start = facet_points[:, 0, :]
+    end = facet_points[:, 1, :]
+
+    # create the vectors for each facet and normalize them to unit vectors
+    facet_vectors = end - start
+    lengths = np.sqrt(np.einsum('ij,ij->i', facet_vectors, facet_vectors))
+    if np.any(lengths == 0.0):
+        # if a length is zero, then a facet had the same starting and ending
+        # point
+        facet_indices = np.nonzero(lengths == 0)[0]
+        offending_facets = facet_points[facet_indices]
+        facet_point_indices = np.array(facets)[facet_indices]
+        message = ''
+        for index, facet, point_indices in zip(
+            facet_indices, offending_facets, facet_point_indices
+        ):
+            message += (
+                f'\nFacet {index} with points {point_indices.tolist()} at '
+                f'{facet.tolist()}'
+            )
+        raise GeometryError(
+            f'{offending_facets.shape[0]} facets have the same starting and '
+            f'ending point:{message}'
+        )
+    unit_vectors = np.divide(facet_vectors, lengths.reshape(-1, 1))
+
+    # vectors from each point to check to the start and end of each facet
+    vec1 = np.array(check_points).reshape(1, -1, 2) - start.reshape(-1, 1, 2)
+    vec2 = np.array(check_points).reshape(1, -1, 2) - end.reshape(-1, 1, 2)
+
+    # signed parallel distance components
+    signed_parallel_vec1 = np.einsum('ijk,ik->ij', -vec1, unit_vectors)
+    signed_parallel_vec2 = np.einsum('ijk,ik->ij', vec2, unit_vectors)
+
+    # clamped parallel distance
+    clamped_distances = np.maximum.reduce(
+        [
+            signed_parallel_vec1,
+            signed_parallel_vec2,
+            np.zeros_like(signed_parallel_vec1),
+        ]
+    )
+
+    # calculate the cross product to get the perpendicular distance
+    perpendicular_distances = np.cross(vec1, unit_vectors.reshape(-1, 1, 2))
+
+    # get the minimum clamped distance
+    minimum_distances = np.hypot(clamped_distances, perpendicular_distances)
+
+    # make sure that the closest distance is not within the allowed tolerance
+    if np.min(minimum_distances) < tolerance:
```

Comment:
I think it would be best to use an absolute and relative tolerance here (e.g. see the zip_points() method).

Reply:
Actually on second thoughts we could probably stay with absolute tolerance as zipping nodes is a bit of a different process where the user might have very large dimensions and what nodes zipped which are 'relatively' close. I think probably in this case we can get away with just an absolute tolerance as currently programmed. Maybe change tolerance to 1e-8 to keep consistency with the other methods?

Reply:
Yup, I should have looked around for other tolerance values in the code - I agree that it should be consistent.  Also, I agree that only an absolute tolerance is needed here, I think.  Maybe I'll rename it to be more clear too or follow whatever other tolerance names that are in the code base.